src/client/Client.cc

```diff
@@ -3066,6 +3066,13 @@ void Client::handle_lease(const MConstRef<MClientLease>& m)
 
 void Client::put_inode(Inode *in, int n)
 {
+  bool locked = false;
+
+  if (!ceph_mutex_is_locked_by_me(client_lock)) {
```

Comment:
`ceph_mutex_is_locked_by_me` is only safe for assertions. It is optimized out without debugging.

Reply:
Okay, then is there any other alternative for this ?

Without this we must always be sure that the `InodeRef in;` declared in the client_lock scope. But to break client_lock there will be many places will do like:

func() {
    client_lock.lock();
    InodeRef in = &inode_map.at(id);  // The inode_map will be under client_lock
    client_lock.unlock();
    std::scoped_lock sl = {in->inode_lock};  // For each inode will has its own inode_lock
    ....
}

Then the in InodeRef's desctruction, it will call the put_inode() which will out of the client_lock scope.

Reply:
Oh, maybe we should switch to use client_lock.is_locked_by_me(), right.